pkg/git/libgit2/managed/init.go

```diff
@@ -75,8 +75,9 @@ func InitManagedTransport(log logr.Logger) error {
 			return
 		}
 
-		err = registerManagedSSH()
-		enabled = true
+		if err = registerManagedSSH(); err != nil {
+			enabled = true
+		}
 	})
 
 	return err
```

Comment:
Given that all subsequent calls are no-ops. I wonder whether we want to keep the err and return it on the future calls.
Otherwise first call you get `false` and a meaningful error, and the next ones just `false`.

Reply:
Why would we want to keep the error around? This is supposed to be a one time thing, right?

Reply:
Ideally this would be called once within the application lifecycle. But it may be called multiple times (i.e. as we do in tests). 
But going through all the current calls, we are completely ignoring the return value.

We can leave as is, as this probably will be changed as part of the unmanaged transport sunsetting.